### PR TITLE
feat: dockerise frontend

### PR DESCRIPTION
**Pull request by zawan-ila**
### Summary

This PR adds support for running the application inside Docker

### Instructions

1. Clone this repo `git clone https://github.com/zawan-ila/arbicoin-frontend.git`
2. Clone the backend `git clone https://github.com/zawan-ila/arbicoin.git`
3. Inside both cloned repos checkout the container branch `git checkout remotes/origin/container`
4. Inside the backend run `docker-compose -f docker_compose_dev.yml --verbose up -d --build`
5. Your application is live at `localhost:3000`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+COPY . .
+
+RUN npm install
+
+RUN npm run build
+
+FROM nginx
+
+COPY --from=builder /app/build /usr/share/nginx/html
+COPY nginx.conf /etc/nginx/conf.d/default.conf
```
Comment (azanbinzahid) [inline Dockerfile:18]: optional: you can make a separate file for nginx

File: nginx.conf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+server {
+ listen 80;
+
+ location / {
+   root /usr/share/nginx/html/;
+   include /etc/nginx/mime.types;
+   try_files $uri $uri/ /index.html;
```
Comment (azanbinzahid) [inline nginx.conf:7]: I hope you understand in depth what is going on here.

**Review by azanbinzahid — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (zawan-ila) [reply Dockerfile:18]: but I need to use the build files from Node Frontend...not sure how I can do that with separate files

File: nginx.conf (hunk 2 above)
Comment (zawan-ila) [reply nginx.conf:7]: have an idea but definitely not in depth...will look into it..

**Pull request merged**